### ISSUE TITLE
Screen reader accessibility

### ISSUE THREAD

**Issue by angelikatyborska** [elixir-lang/ex_doc#1756]
As promised in https://github.com/elixir-lang/ex_doc/pull/1749, I also took a stab at testing the screen reader accessibility of HexDocs and would love to submit a PR soon to help improve things. Here are my notes with potential improvements / bugs to fix. Disclaimer: I don't consider myself to be an accessibility expert, I just learned how to use VoiceOver and automated tools like Wave to test stuff. 

- Missing landmark elements: `<nav>`, `<main>`
- Expand menu button is missing `aria-controls` and `aria-expanded` properties
- Search input: `<p>` is not a valid child of `<label>`
- Version select:
	- Lacks a label.
	- Decorative character "\25bc" in version select is being read as "down pointing black pointer".
- Expand buttons in navigation links are missing `aria-controls` and `aria-expanded` properties.
- Consider using aria-current for links visually styled as "active".
- Settings modal:
	- Close button lacks a label, being read as "times button".
	- Theme select lacks a label.
- Copy button for code snippets:
	- Lacks aural feedback after successful copying, consider `aria-live` on the "copied!" text.
- Search HexDocs modal: close button lacks a label, being read as "times button".
- Consider a "skip to main content" button to quickly bypass the navigation entirely.

**Comment by josevalim** [comment]
100% welcome and extremely appreciated. :) Thank you!

**Comment by josevalim** [comment]
Closing in favor of the PR :)
